File: cs101-Algorithms_and_ADTs/asg2-C-List_ADT/ListTest.c
/* $Id: ListTest.c,v 1.2 2010-10-17 08:48:47-07 - - $
 * Derek Frank, dmfrank@ucsc.edu
 *
 * NAME
 *   ListTest -- used to test List.c
 * 
 * DESCRIPTION
 *   Tests List.c to check if it functions properly and according to specifications.
 */

#include <stdio.h>
#include <stdlib.h>
#include "List.h"

int main (int argc, const char * argv[]) {
	ListRef A = newList();
	ListRef B = newList();
	int i;
	
	for ( i=1; i<=10; ++i) {
		insertAfterBack(A, i);
		insertAfterBack(B, 11-i);
	}
	printf("A = ");
	printList(A);
	printf("\nB = ");
	printList(B);
	
	for( i=1; i<=6; ++i ) {
		insertAfterBack(A, getFront(B));
		deleteFront(B);
	}
	printf("\nA = ");
	printList(A);
	printf("\nB = ");
	printList(B);
	ListRef C = copyList(A);
	printf("\nC = ");
	printList(C);
	
	if ( equals(A,B) ) { printf("\nA equals B"); }
	else { printf("\nA does not equal B"); }
	if ( equals(A,C) ) { printf("\nA equals C"); }
	else { printf("\nA does not equal C"); }
	
	ListRef D = catList(A,B);
	printf("\nD = ");
	printList(D);
	if ( equals(A,D) ) {
		printf("\nA equals D");
	}else {
		printf("\nA does not equal D");
	}
	if ( equals(D,A) ) { printf("\nD equals A"); }
	else { printf("\nD does not equal A"); }
	
	int lengthCompare = ( getLength(A) == getLength(D) );
	printf("\nLength A = %d and Length D = %d and Length C = %d", getLength(A), getLength(D),getLength(C));
	printf("\nLength Compare = %d\n",lengthCompare);
	
	makeEmpty(D);
	printList(D);
	freeList(&D);
	printList(D);
	return (0);
}

File: cs101-Algorithms_and_ADTs/asg1-Java-List_ADT/ListTest.java
/* $Id: ListTest.java,v 1.1 2010-10-10 04:36:14-07 - - $
 * Derek Frank, dmfrank@ucsc.edu
 * 
 * NAME
 *   ListTest -- used to test List.java.
 * 
 * DESCRIPTION
 *   Tests List.java to check if it functions properly and according to specifications.
 */

class ListTest {
  public static void main (String[] args) {
    List A = new List();
    List B = new List();
    
    for ( int i=1; i<=10; ++i) {
      A.insertAfterLast(i);
      B.insertAfterLast(11-i);
    }
    System.out.println("A = " + A);
    System.out.println("B = " + B);

    for(int i=1; i<=6; ++i){
      A.insertAfterLast(B.getFirst());
      B.deleteFirst();
    }
    System.out.println("A = " + A);
    System.out.println("B = " + B);
    List C = A.copy();
    System.out.println("C = " + C);
    System.out.println("A " + (A.equals(B)?"equals":"does not equal") + " B");
    System.out.println("A " + (A.equals(C)?"equals":"does not equal") + " C");
    
    List D = A.cat(B);
    System.out.println("D = " + D);
    System.out.println("A " + (A.equals(D)?"equals":"does not equal") + " D");
    System.out.println("D " + (D.equals(A)?"equals":"does not equal") + " A");
    boolean lengthCompare = ( A.getLength() == D.getLength() );
    System.out.println("Length A = " + A.getLength() + " and Length D = " + D.getLength());
    System.out.println("Length Compare = " + lengthCompare);
    
    
  }
}